File: Java/Medium/Accepted/making-file-names-unique_6.java
class Solution {
    public String[] getFolderNames(String[] names) {
        
        String[] res = new String[names.length];
        Map<String, Integer> map = new HashMap<>();
        for (int i = 0; i < names.length; i++) {
            String name = names[i];
            if (!map.containsKey(name)) {
                res[i] = name;
                map.put(name, 1);
            } else {
                int k = map.get(name);
                while (map.containsKey(name + "(" + k + ")")) {
                    k++;
                }
                name = name + "(" + k + ")";
                map.put(name, 1);
                map.put(names[i], k + 1);
                res[i] = name;
            }
        }
        return res;
    }
}

File: C++/Medium/Accepted/making-file-names-unique_6.cpp
class Solution {
public:
    vector<string> getFolderNames(vector<string>& names) {
        
        vector<string> ans;
        unordered_map<string, int> m;
        for (auto n : names) {
            if (m.find(n) == m.end()) {
                ans.push_back(n);
                m[n] = 1;
            } else {
                int k = m[n];
                string newn = n + "(" + to_string(k) + ")";
                while (m.find(newn) != m.end()) {
                    ++k;
                    newn = n + "(" + to_string(k) + ")";
                }
                ans.push_back(newn);
                m[n] = k + 1;
                m[newn] = 1;
            }
        }
        return ans;
    }
};